getMimeTypeForFile does not handle files without extensions correctly

https://github.com/kiwix/kiwix-tools/blob/f99b6965e99f73f41de73ea32aed478888db7275/src/server/kiwix-serve.cpp#L133

This is incorrect on 64-bit platforms, since the index returned by all `find` functions is of type `size_t` there. So, it can overflow. But worse yet is the next line:

https://github.com/kiwix/kiwix-tools/blob/f99b6965e99f73f41de73ea32aed478888db7275/src/server/kiwix-serve.cpp#L135

Since `std::string::npos` is also `size_t`, the sentinel value is usually out of bounds for 32-bit. Since comparing 32-bit integer to 64-bit will result in the former being implicitly promoted to 64-bit, this means that the comparison is simply never true.

This is reflected by a warning produced when building with Clang:
```
../src/server/kiwix-serve.cpp:135:11: error: result of comparison of constant
18446744073709551615 expression of type 'unsigned int' is always true
[-Werror,-Wtautological-constant-out-of-range-compare]
  if (pos != std::string::npos) {
      ~~~ ^  ~~~~~~~~~~~~~~~~~
```

--- a/src/server/kiwix-serve.cpp
+++ b/src/server/kiwix-serve.cpp
@@ -130,7 +130,7 @@ get_from_humanReadableBookId(const std::string& humanReadableBookId) {
 static std::string getMimeTypeForFile(const std::string& filename)
 {
   std::string mimeType = "text/plain";
-  unsigned int pos = filename.find_last_of(".");
+  auto pos = filename.find_last_of(".");
 
   if (pos != std::string::npos) {
     std::string extension = filename.substr(pos + 1);

